### PR TITLE
Adds unit tests and improves BloggingRemindersScheduler.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Adds unit tests to `BloggingRemindersScheduler`.

The `Schedule` now implements `Comparable` and `Equatable`.  This effectively means that the schedule is now can be compared and doesn't care about the ordering of the elements to know when two schedules are the same.

## To test:

Since the scheduler isn't wired and testable from within the App yet, just evaluate the changes visually and make sure the unit tests succeed in running.

## Regression Notes

1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+protocol NotificationScheduler {
+    func add(_ request: UNNotificationRequest, withCompletionHandler completionHandler: ((Error?) -> Void)?)
+    func removePendingNotificationRequests(withIdentifiers identifiers: [String])
+}
+
+extension UNUserNotificationCenter: NotificationScheduler {
+}
+
+/*
+class Test: UNUserNotificationCenter {
```
Comment (frosty) [inline WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift]: Can this be removed?

File: WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift [hunk 2]
```diff
@@ -54,7 +84,7 @@ class BloggingRemindersScheduler {
 
     /// The notification center
```
Comment (frosty) [inline WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift]: We should update this comment to match

File: WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift [hunk 3]
```diff
@@ -40,6 +66,10 @@ class BloggingRemindersScheduler {
         /// The default reminder hour.  In the future we may want to replace this constant with a more customizable approach.
         ///
         static let defaultHour = 10
+
+        static func < (lhs: BloggingRemindersScheduler.Weekday, rhs: BloggingRemindersScheduler.Weekday) -> Bool {
```
Comment (frosty) [inline WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift:63]: Do we need to implement `==` here as well?

**Review by frosty — APPROVED** [review]
This looks great, and tests pass. Just left a few small code comments, otherwise :shipit:

File: WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift (hunk 1 above)
Comment (diegoreymendez) [reply WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift]: Yes

File: WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift (hunk 3 above)
Comment (diegoreymendez) [reply WordPress/Classes/Utility/Blogging Reminders/BloggingRemindersScheduler.swift:63]: Swift default implementation of `Equatable` for enums covers what we need.

You can try this in a playground:

```swift
enum Test {
    case zero
    case uno
}

let first: Test = .zero
let second: Test = .uno
let third: Test = .uno

print(first == second) // false
print(first == third) // false
print(second == third) // true
```

**Comment by diegoreymendez** [comment]
Hey @frosty, this PR is now ready for another look!

**Review by frosty — APPROVED** [review]

**Pull request merged**